src/lib/authentication/arkBaseQuery.ts

```diff
@@ -0,0 +1,53 @@
+import { fetchBaseQuery } from "@reduxjs/toolkit/query";
+import type {
+  BaseQueryFn,
+  FetchArgs,
+  FetchBaseQueryError,
+} from "@reduxjs/toolkit/query";
+import { authProvider } from "../../features/authentication/authenticationSlice";
+import {
+  tokenReceived,
+  loggedOut,
+} from "../../features/authentication/authenticationSlice";
+import { RootState } from "../../app/configureStore";
+
+function GetBaseQuery(baseUrl: string) {
+  return fetchBaseQuery({
+    baseUrl: baseUrl,
+    prepareHeaders: (headers, { getState }) => {
+      const token = (getState() as RootState).auth.data?.token;
+
+      // If we have a token set in state, let's assume that we should be passing it.
+      if (token) {
+        headers.set("authorization", `Bearer ${token}`);
+      }
+
+      return headers;
+    },
+  });
+}
+
+export function GetArkReauthQuery(baseUrl: string) {
+  const baseQueryWithReauth: BaseQueryFn<
+    string | FetchArgs,
+    unknown,
```

Comment:
I don't like this. you also expect a 'dispact' method to be present below.

Reply:
what do you mean?


ps: this is missing an important part that me and Marco already implemented for another project that handles dynamic urls and paths

Reply:
the 'unknown' type for 'api' and then expecting api to have some properies.

I tolerate 'unknown' when is a passthrough and not used.